### PR TITLE
28 support different signal types

### PR DESCRIPTION
**Pull request by medeaohm**

### CODE REVIEW

File: app/src/main/java/org/helpapaw/helpapaw/signalsmap/SignalsMapPresenter.java [hunk 1]
```diff
@@ -28,12 +28,15 @@
     private int radius;
     private int timeout;
 
+    private boolean[] selection;
```
Comment (BurningAXE) [inline app/src/main/java/org/helpapaw/helpapaw/signalsmap/SignalsMapPresenter.java:31]: I'd rename this 'selectedTypes' to be more clear

File: app/src/main/java/org/helpapaw/helpapaw/signalsmap/SignalsMapPresenter.java [hunk 2]
```diff
@@ -28,12 +28,15 @@
...
+
     private double currentMapLatitude;
     private double currentMapLongitude;
 
     private String photoUri;
     private boolean sendSignalViewVisibility;
     private List<Signal> signalsList;
+//    private List<String> selectedTypes;
```
Comment (BurningAXE) [inline app/src/main/java/org/helpapaw/helpapaw/signalsmap/SignalsMapPresenter.java:39]: Guess that should be deleted

File: app/src/main/java/org/helpapaw/helpapaw/signalsmap/SignalsMapPresenter.java [hunk 3]
```diff
@@ -256,14 +262,15 @@ public void onSignalInfoWindowClicked(Signal signal) {
     public void onBackButtonPressed() {
         if (sendSignalViewVisibility) {
             setSendSignalViewVisibility(false);
+//            setFilterSignalViewVisibility(false);
```
Comment (BurningAXE) [inline app/src/main/java/org/helpapaw/helpapaw/signalsmap/SignalsMapPresenter.java:265]: And probably this one :)

File: app/src/main/java/org/helpapaw/helpapaw/signalsmap/SignalsMapPresenter.java [hunk 4]
```diff
@@ -333,6 +340,14 @@ private void setSendSignalViewVisibility(boolean visibility) {
         getView().setAddSignalViewVisibility(visibility);
     }
 
+//    public void setSelectedTypes(List<String> selectedTypes) {
+//        this.selectedTypes = selectedTypes;
+//    }
```
Comment (BurningAXE) [inline app/src/main/java/org/helpapaw/helpapaw/signalsmap/SignalsMapPresenter.java:345]: Aaand this one :)

File: app/src/main/java/org/helpapaw/helpapaw/utils/MultiSelectionSpinner.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+import androidx.appcompat.app.AlertDialog;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MultiSelectionSpinner extends androidx.appcompat.widget.AppCompatSpinner implements
+        DialogInterface.OnMultiChoiceClickListener {
+
+    private static final String TAP_TO_FILTER = "Tap to filter";
```
Comment (BurningAXE) [inline app/src/main/java/org/helpapaw/helpapaw/utils/MultiSelectionSpinner.java:17]: This should be in strings.xml so it can be translated

**Comment by medeaohm** [comment]
Currently the following (incorrect?) behavior is observed:
1. The user opens the filter view
2. The user makes a selection1
3. The user presses the Filter button
-----> signals are filtered
4. The user re-opens the filter view - selection1 is visible
5. The user change selection to selection2
6. The user presses the back button
-----> The signals on the map are still filtered using selection1
7. The user re-opens the filter view - selection2 is visible

**Review by BurningAXE — APPROVED** [review]

**Comment by BurningAXE** [comment]
Changes are included in #31 so this one is just closed.

**Pull request closed without merging**